### PR TITLE
activityCategory should not be static / should be overwritten

### PR DESCRIPTION
**Pull request by narciszait**
The category property of an UIActivity is one of the 5 properties that can be overwritten when subclassing UIActivity (https://developer.apple.com/documentation/uikit/uiactivity)
As it is today, Category cannot be overwritten, unlike all the other properties that are exposed to Xamarin-iOS

The workaround i have is to use it like this:
`public static new UIActivityCategory Category { [Export("activityCategory")] get => UIActivityCategory.Share; }`

unlike the other ones:

`public override string Title => "Activity title";`
`public override NSString Type => new NSString("custom activity type");`

### CODE REVIEW

File: src/uikit.cs [hunk 1]
```diff
@@ -1560,7 +1560,7 @@ interface UIActivity {
 		[Export ("activityDidFinish:")]
 		void Finished (bool completed);
 
-		[Export ("activityCategory"), Static]
+		[Export ("activityCategory")]
```
Comment (mandel-macaque) [inline src/uikit.cs:1563]: Thanks for the bug fix! While the change is valid in terms of the API usage, this adds an API breaking change. That means, our users will have compilation issues when they receive an update. 

The correctly way to do this is to add a deprecation, add to a small #if dance to ensure that when we have a chance to break the API, we can do so.

You can see an example here: https://github.com/xamarin/xamarin-macios/blob/main/src/uikit.cs#L9074

So we could need to do something like the following (please check that it compiles etc..)
```csharp

#if XAMCORE_5_0
		[Export ("activityCategory")]
#else
		[Obsolete ("Use the 'Category2' property instead.")]
		[Export ("activityCategory"), Static]
#endif
		UIActivityCategory Category { get; }

#if !XAMCORE_5_0
		[Export ("activityCategory")]
		UIActivityCategory Category2 { get; }
#endif
```

This way, we leave the old property and won't break our clients apps, they'll get a new warning and will fix accordingly. Next time we are allowed to change the API, we will get the correct behaviour.

**Review by mandel-macaque — CHANGES_REQUESTED** [review]

File: src/uikit.cs (hunk 1 above)
Comment (narciszait) [reply src/uikit.cs:1563]: Thanks for the feedback - i will do the changes tomorrow

Comment (narciszait) [reply src/uikit.cs:1563]: i am open to suggestions on the naming 😀

Comment (MarLoe) [reply src/uikit.cs:1563]: Perhaps it could be called `ActivityCategory` 😉

**Comment by rolfbjarne** [comment]
This is incorrect, the [`activityCategory`](https://developer.apple.com/documentation/uikit/uiactivity/1620656-activitycategory?language=objc) property is a `class` property in Objective-C, which means that making it static in C# is correct.

Note that in Objective-C it's possible to override static (class) members, while in C# it's not, so there's no easy way to map overridable class (static) Objective-C members to C#.

In fact, your workaround is the actual solution, you have to define the static member yourself.

**Comment by rolfbjarne** [comment]
In fact I believe you can test this yourselve by defining an instance property like this:

```cs
public new UIActivityCategory Category { [Export("activityCategory")] get => UIActivityCategory.Share; }
``

I'm guessing this won't work (and this would be the equivalent of what this PR would do).

**Pull request closed without merging**